controllers/super_stream_controller.go

```diff
@@ -0,0 +1,183 @@
+	"github.com/rabbitmq/messaging-topology-operator/internal"
+	"github.com/rabbitmq/messaging-topology-operator/internal/managedresource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	clientretry "k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strconv"
+)
+
+// SuperStreamReconciler reconciles a RabbitMQ Super Stream, and any resources it comprises of
+type SuperStreamReconciler struct {
+	client.Client
+	Log                   logr.Logger
+	Scheme                *runtime.Scheme
+	Recorder              record.EventRecorder
+	RabbitmqClientFactory internal.RabbitMQClientFactory
+}
+
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=exchanges,verbs=get;create;update;patch;delete
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=queues,verbs=get;create;update;patch;delete
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=bindings,verbs=get;create;update;patch;delete
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=superstreams,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=superstreams/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=rabbitmqclusters,verbs=get;list;watch
+// +kubebuilder:rbac:groups=rabbitmq.com,resources=rabbitmqclusters/status,verbs=get
+// +kubebuilder:rbac:groups="",resources=services,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",resources=events,verbs=get;create;patch
+
+func (r *SuperStreamReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	superStream := &topologyv1alpha1.SuperStream{}
+	if err := r.Get(ctx, req.NamespacedName, superStream); err != nil {
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	rmq, _, _, err := internal.ParseRabbitmqClusterReference(ctx, r.Client, superStream.Spec.RabbitmqClusterReference, superStream.Namespace)
+	if err != nil {
+		return handleRMQReferenceParseError(ctx, r.Client, r.Recorder, superStream, &superStream.Status.Conditions, err)
+	}
+
+	logger.Info("Start reconciling")
+
+	if superStream.Spec.Partitions < len(superStream.Status.Partitions) {
+		// This would constitute a scale down, which may result in data loss.
+		err := fmt.Errorf(
+			"SuperStreams cannot be scaled down: an attempt was made to scale from %d partitions to %d",
+			len(superStream.Status.Partitions),
+			superStream.Spec.Partitions,
+		)
+		msg := fmt.Sprintf("SuperStream %s failed to reconcile", superStream.Name)
+		logger.Error(err, msg)
+		if writerErr := r.SetReconcileSuccess(ctx, superStream, topology.NotReady(msg, superStream.Status.Conditions)); writerErr != nil {
+			logger.Error(writerErr, failedStatusUpdate, "status", superStream.Status)
+		}
+		return reconcile.Result{}, err
```

Comment:
Returning the error is going to trigger another recocile loop, which will never succeed unless the user takes action and changes the partitions.

I would be inclined to return `Result{}, nil` instead, perhaps produce an event, on top the setting the reconcile success condition to false.